explorer/src/lib/components/block-details/BlockDetails.svelte

```diff
@@ -0,0 +1,170 @@
+<svelte:options immutable={true} />
+
+<script>
+  import { mdiArrowLeft, mdiArrowRight } from "@mdi/js";
+  import { AppAnchor, ListItem } from "$lib/components";
+  import { Card, Icon, ProgressBar, Switch } from "$lib/dusk/components";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import {
+    calculateAdaptiveCharCount,
+    getRelativeTimeString,
+    middleEllipsis,
+  } from "$lib/dusk/string";
+  import { onMount } from "svelte";
+  import "./BlockDetails.css";
+
+  const formatter = createCurrencyFormatter("en", "DUSK", 0);
+  const blockHeight = 34526;
+
+  /** @type {number} */
+  let screenWidth = window.innerWidth;
+
+  /** @type {boolean} */
+  let togglePayload = false;
+
+  onMount(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      const entry = entries[0];
+
+      screenWidth = entry.contentRect.width;
+    });
+
+    resizeObserver.observe(
+      document.body.getElementsByClassName("details-list__definition")[0]
+    );
+
+    return () => resizeObserver.disconnect();
+  });
+</script>
+
+<Card className="block-details">
+  <header slot="header" class="block-details__header">
+    <h3 class="block-details__header-heading">
+      Block Details <span class="block-details__header-heading--block-height">
```

Comment:
Could you please point me to the rule you have followed to compose this BEM class name?

Reply:
are you serious here?

Reply:
If there is an issue with the naming please say there is an issue and suggest a more appropriate way.

Reply:
I am serious, yes. After the conversations around the Buttons in the past days, it seems we are following a BEM variation I am not aware of. I have checked the [resource](https://en.bem.info/methodology/quick-start/#element) that we have been referencing last week, I personally can't see this being correct, however, I might also be missing something and wanted to make sure I am not missing something before saying it's wrong.

Reply:
Now that I'm looking the the BEM methodology, i can say that the class name follows the main rule.
`block-details` - block name
`header-heading` - element-name
`block-height` - modifier value

Reply:
The modifier is used to `change appearance, behavior or state`.

For example, if you have menu items, each would have class of `nav__item` and the currenrtly active one will have the modifier of `--active` (so `nav__item--active`, in addition to the `nav__item`). In this case, the `nav__item` class will provide the default styling for all nav bar items, whereas the `nav__item--active` will apply the appearance of the active state to the currently active item.


In yours case, the span is actually another element. So the class name can just be `block-details__block-height`.

Essentially, there shouldn't be a modifier applied without a block or element class before that, as the modifier modifies the block or the element.